### PR TITLE
mds: revert standby-replay trimming changes

### PR DESCRIPTION
**Pull request by batrick**
I want to do some testing in Linode with this change-set to see if I can reproduce the issues with up:rejoin phase coming from standby-replay [1]. So leaving this as a draft for now.

Once merged I think this should also cook in main's QA for a month or so before we do any backports.

[1] https://tracker.ceph.com/issues/40213

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by batrick** [comment]
cc @lxbsz

**Comment by lxbsz** [comment]
> cc @lxbsz

This looks reasonable and good to me.

File: src/mds/MDCache.cc [hunk 1]
```diff
@@ -6819,19 +6819,15 @@ std::pair<bool, uint64_t> MDCache::trim_lru(uint64_t count, expiremap& expiremap
...
     if (!dn) {
       break;
     }
-    if (is_standby_replay && dn->get_linkage()->inode) {
-      // we move the inodes that need to be trimmed to the end of the lru queue.
-      // refer to MDCache::standby_trim_segment
-      lru.lru_insert_bot(dn);
-      break;
+    if (is_standby_replay && dn->get_linkage()->inode && dn->get_linkage()->inode->item_open_file.is_on_list()) {
+      dout(20) << "unexpirable: " << *dn << dendl;
```
[the remark this answers is not in the document]
Comment (batrick) [reply src/mds/MDCache.cc:6831]: ... no! Great catch. I don't know why I did that. Must be a mistake. I've done a proper `git revert` of the two commits, squashed them, and added this debug message + removed a minor conflict.... just to ensure I didn't miss anything else.

**Comment by batrick** [comment]
jenkins test make check

**Review by vshankar — APPROVED** [review]

**Comment by vshankar** [comment]
@batrick I see you added your testing tag - do you plan to run this through tests anytime soon or I can pick it up...

**Comment by vshankar** [comment]
@batrick planning to run this through tests anytime soon?

**Comment by batrick** [comment]
I still plan to... waiting for sepia to stabilize : /

**Comment by vshankar** [comment]
> I still plan to... waiting for sepia to stabilize : /

@batrick Do you plan to pick this up anytime soon?

**Comment by vshankar** [comment]
> > I still plan to... waiting for sepia to stabilize : /
> 
> @batrick Do you plan to pick this up anytime soon?

@batrick - any update on this?

**Comment by vshankar** [comment]
@ifed01 FYI - this is being run through fs suite by @batrick

**Comment by vshankar** [comment]
@batrick Do you have an update on the tests?

**Comment by batrick** [comment]
Not yet, I ran this through QA. Will update when I have time to go through the results.

**Comment by batrick** [comment]
jenkins test make check

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by kotreshhr** [comment]
looks good

**Comment by rishabh-d-dave** [comment]
@batrick Do you want me to test this PR?

**Comment by batrick** [comment]
> @batrick Do you want me to test this PR?

You may.

**Comment by vshankar** [comment]
> > @batrick Do you want me to test this PR?
> 
> You may.

On it this week.

**Comment by vshankar** [comment]
Builds in progress - https://shaman.ceph.com/builds/ceph/wip-vshankar-testing-20231107.042705/

**Comment by vshankar** [comment]
Tests running here - https://pulpito.ceph.com/vshankar-2023-11-07_06:04:04-fs-wip-vshankar-testing-20231107.042705-testing-default-smithi/

**Comment by vshankar** [comment]
> Tests running here - https://pulpito.ceph.com/vshankar-2023-11-07_06:04:04-fs-wip-vshankar-testing-20231107.042705-testing-default-smithi/

Test run looks good. I'll prepare the run wiki and merge this soon. Nice work @batrick

**Review by vshankar — APPROVED** [review]
https://tracker.ceph.com/projects/cephfs/wiki/Main#29-Nov-2023

**Comment by microyahoo** [comment]
hi @vshankar, @batrick, would you plan to backport this PR to pacific? I have met this issue frequently.

**Comment by vshankar** [comment]
> hi @vshankar, @batrick, would you plan to backport this PR to pacific? I have met this issue frequently.

Hi @microyahoo - pacific is EOL'd. See: https://docs.ceph.com/en/latest/releases/pacific/#v16-2-15-pacific

You should consider upgrading.

**Comment by microyahoo** [comment]
> > hi @vshankar, @batrick, would you plan to backport this PR to pacific? I have met this issue frequently.
> 
> Hi @microyahoo - pacific is EOL'd. See: https://docs.ceph.com/en/latest/releases/pacific/#v16-2-15-pacific
> 
> You should consider upgrading.

hi @vshankar, thanks for your kindly reminder,  I will upgrade to latest.

**Pull request merged**